[CI] SingleValueQueryTests testMatchSome failing

Multiple tests failing:
SingleValueQueryTests » testMatchSome {p0=StandardSetup[fieldType=float, multivaluedField=true, empty=false, count=100]}FAILED
SingleValueQueryTests » testMatchSome {p0=StandardSetup[fieldType=short, multivaluedField=true, empty=true, count=100]}FAILED
SingleValueQueryTests » testMatchSome {p0=StandardSetup[fieldType=long, multivaluedField=true, empty=true, count=100]}FAILED
SingleValueQueryTests » testMatchSome {p0=StandardSetup[fieldType=double, multivaluedField=true, empty=false, count=100]}FAILED
SingleValueQueryTests » testMatchSome {p0=StandardSetup[fieldType=keyword, multivaluedField=true, empty=false, count=100]}FAILED	
SingleValueQueryTests » testMatchSome {p0=StandardSetup[fieldType=long, multivaluedField=true, empty=false, count=100]}FAILED

**Build scan:**
https://gradle-enterprise.elastic.co/s/7nmrltng2r42c/tests/:x-pack:plugin:esql:test/org.elasticsearch.xpack.esql.querydsl.query.SingleValueQueryTests/testMatchSome%20%7Bp0=StandardSetup%5BfieldType=float,%20multivaluedField=true,%20empty=false,%20count=100%5D%7D


**Reproduction line:**
```
./gradlew ':x-pack:plugin:esql:test' --tests "org.elasticsearch.xpack.esql.querydsl.query.SingleValueQueryTests.testMatchSome {p0=StandardSetup[fieldType=float, multivaluedField=true, empty=false, count=100]}" -Dtests.seed=F38F8AC7FC738DC1 -Dtests.configure_test_clusters_with_one_processor=true -Dtests.locale=el-CY -Dtests.timezone=Africa/Nairobi -Druntime.java=21
```

**Applicable branches:**
main

**Reproduces locally?:**
Didn't try

**Failure history:**
[Failure dashboard for `org.elasticsearch.xpack.esql.querydsl.query.SingleValueQueryTests#testMatchSome {p0=StandardSetup[fieldType=float, multivaluedField=true, empty=false, count=100]}`](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testMatchSome%20%7Bp0%3DStandardSetup%5BfieldType%3Dfloat,%20multivaluedField%3Dtrue,%20empty%3Dfalse,%20count%3D100%5D%7D'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.esql.querydsl.query.SingleValueQueryTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))


**Failure excerpt:**
```
java.lang.AssertionError: no warnings, expected: [WARN: Line -1:-1: evaluation of [] failed, treating result as null. Only first 20 failures recorded., WARN: Line -1:-1: java.lang.IllegalArgumentException: single-value function encountered multi-value]

  at __randomizedtesting.SeedInfo.seed([F38F8AC7FC738DC1:27291E8CB6366312]:0)
  at org.junit.Assert.fail(Assert.java:88)
  at org.junit.Assert.assertTrue(Assert.java:41)
  at org.junit.Assert.assertNotNull(Assert.java:712)
  at org.elasticsearch.test.ESTestCase.assertWarnings(ESTestCase.java:627)
  at org.elasticsearch.test.ESTestCase.assertWarnings(ESTestCase.java:596)
  at org.elasticsearch.xpack.esql.querydsl.query.SingleValueQueryTests.runCase(SingleValueQueryTests.java:168)
  at org.elasticsearch.xpack.esql.querydsl.query.SingleValueQueryTests.lambda$testMatchSome$0(SingleValueQueryTests.java:86)
  at org.elasticsearch.xpack.esql.querydsl.query.SingleValueQueryTests.testCase(SingleValueQueryTests.java:188)
  at org.elasticsearch.xpack.esql.querydsl.query.SingleValueQueryTests.testMatchSome(SingleValueQueryTests.java:82)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
  at java.lang.reflect.Method.invoke(Method.java:580)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1583)

```

--- a/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/querydsl/query/SingleValueQueryTests.java
+++ b/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/querydsl/query/SingleValueQueryTests.java
@@ -156,15 +156,18 @@ public class SingleValueQueryTests extends MapperServiceTestCase {
         int expected = 0;
         int min = docsStart != null ? docsStart : 0;
         int max = docsStop != null ? docsStop : fieldValues.size();
+        int valuesCount = 0;
         for (int i = min; i < max; i++) {
-            if (fieldValues.get(i).size() == 1) {
+            int mvCount = fieldValues.get(i).size();
+            if (mvCount == 1) {
                 expected++;
             }
+            valuesCount += mvCount;
         }
         assertThat(count, equalTo(expected));
 
         // query's count runs against the full set, not just min-to-max
-        if (fieldValues.stream().anyMatch(x -> x.size() > 1)) {
+        if (valuesCount > 0 && fieldValues.stream().anyMatch(x -> x.size() > 1)) {
             assertWarnings(
                 "Line -1:-1: evaluation of [] failed, treating result as null. Only first 20 failures recorded.",
                 "Line -1:-1: java.lang.IllegalArgumentException: single-value function encountered multi-value"

